src/pages/MyProfile.jsx

```diff
@@ -1,15 +1,240 @@
-import React from "react";
+import React, {useEffect, useState} from "react";
+import PopupJS from "reactjs-popup";
 import PageLayout from "../components/PageLayout";
 import Header from "../components/Header";
 import Footer from "../components/Footer";
+import api from "../http/api";
+
+import 'reactjs-popup/dist/index.css';
+import hide from "../assets/hide-pass.svg";
+import show from "../assets/show-pass.svg";
+import {useNavigate, useParams} from "react-router-dom";
+import {toast} from "react-toastify";
 
 function MyProfile() {
+
+    const [profileData, setProfileData] = useState({
+        id: '',
+        username: '',
+        firstName: '',
+        lastName: '',
+        roles: ''
+    });
+
+
+    useEffect(() => {
+        const fetchData = async () => {
+            try {
+                await api.get('/stats/user', {
+                    headers: {
+                        'Authorization': 'Bearer ' + localStorage.getItem('token')
+                    }
+                }).then( (res) => setProfileData(res.data));
+               } catch (error) {}
+        };
+
+        fetchData();
+
+    }, []);
+
+    const sendPasswordReset = async () => {
+        try {
+            await api.post('/auth/reset', {
+                headers: {
+                    'Authorization': 'Bearer ' + localStorage.getItem('token')
+                }
+            });
+        } catch (error) {}
+    };
+
+    const invalidatePatient = async () => {
+        try {
+            await api.delete('/patient/' + profileData.id, {
+                headers: {
+                    'Authorization': 'Bearer ' + localStorage.getItem('token')
+                }
+            });
+        } catch (error) {}
+    };
+
+    const tableRowStyle = {
+        textAlign: 'center',
+        fontSize: 25,
+        padding: '10px',
+        paddingRight: '20px'
+    };
+
+
+    const buttonStyle = {
+        backgroundColor: '#57b0ec',
+        color: '#fff',
+        padding: '8px 12px',
+        border: 'none',
+        cursor: 'pointer',
+        fontSize: 15,
+        fontWeight: '600',
+        borderRadius: '8px'
+    };
+
+    const navigate = useNavigate();
+    const { tkn } = useParams();
+
+    const [showPass, setShowPass] = useState(false);
+    const [showConfirmPass, setShowConfirmPass] = useState(false);
+    const [data, setData] = useState({
+        newPass : '',
+        confirmPass : '',
+        token : tkn
+    });
+
+    const togglePass = () => {
+        setShowPass(prevState => !prevState);
+    }
+
+    const toggleConfirmPass = () => {
+        setShowConfirmPass(prevState => !prevState);
+    }
+    const handleChange = (e) => {
+        const { name, value } = e.target;
+        setData({...data, [name] : value});
+        e.target.focus();
+        console.log("Name: " + name + " , value: " + value);
+    }
+    const handleSubmit = async (e) => {

```

Comment:
Before handling submit and sending data to API check if passwords are valid such as if all three inputs are not blank and if newPass is passing regex match like in registration form.

Reply:
How can I get the current password so that I can compare the oldPass input to it?

Reply:
There is no comparing on client side (frontend), client sends three inputs (oldPass, newPass, confirmPass) and after that server side sends response 200 OK if request is successful and password is changed or 400 BADREQUEST if password is not valid or matching. Server side also checks for strength of the password. Minimum 8 characters and needs to have letters and numbers at least. So basically you will get exception on frontend (error) if request is not good.